src/hooks/useBestV3Trade.ts

```diff
@@ -26,7 +27,10 @@ export function useBestV3Trade(
   const routingAPIEnabled = useRoutingAPIEnabled()
   const isWindowVisible = useIsWindowVisible()
 
-  const [debouncedAmount, debouncedOtherCurrency] = useDebounce([amountSpecified, otherCurrency], 200)
+  const [debouncedAmount, debouncedOtherCurrency] = useDebounce(
+    useMemo(() => [amountSpecified, otherCurrency], [amountSpecified, otherCurrency]),
```

Comment:
interesting catch. This was a pretty bad bug i think.. i wonder if splitting up the debounce in 2 would be more efficient

Reply:
I think this is more efficient, because it doesn't create an extra setTimeout.